tests/test_recall_formatters.py

```
@@ -0,0 +1,643 @@
+    def test_null_metadata_does_not_crash(self):
+        result = _make_result()
+        result["metadata"] = None
+        output = format_human_output([result])
+        assert "UNKNOWN" not in output  # human output doesn't show type in flat mode
+        assert "1 matching learnings" in output
+
+    def test_multiline_content_indented_flat(self):
+        results = [_make_result(content="line1\nline2")]
+        output = format_human_output(results)
+        lines = output.split("\n")
+        content_lines = [l for l in lines if "line1" in l or "line2" in l]
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)